app/models/ems_refresh/refreshers/ems_refresher_mixin.rb

```diff
@@ -52,8 +45,76 @@ def refresh
         _log.info "Refreshing all targets...Complete"
       end
 
-      def save_inventory(ems, _targets, hashes)
-        EmsRefresh.save_ems_inventory(ems, hashes)
+      def preprocess_targets
+        @full_refresh_threshold = options[:full_refresh_threshold] || 10
+
+        # See if any should be escalated to a full refresh
+        @targets_by_ems_id.each do |ems_id, targets|
+          ems = @ems_by_ems_id[ems_id]
+          ems_in_list = targets.any? { |t| t.kind_of?(ExtManagementSystem) }
+
+          if ems_in_list
+            _log.info "Defaulting to full refresh for EMS: [#{ems.name}], id: [#{ems.id}]." if targets.length > 1
+            targets.clear << ems
+          elsif !ems_in_list && targets.length >= @full_refresh_threshold
```

Comment:
the `!ems_in_list` is automatically implied by it not taking the first `if` branch.